roadmap/implementers-guide/src/node/subsystems-and-jobs.md

```diff
@@ -9,3 +9,5 @@ Most work that happens on the Node-side is related to building on top of a speci
 Since this goal of determining when to start and conclude work relative to a specific relay-parent is common to most, if not all subsystems, it is logically the job of the Overseer to distribute those signals as opposed to each subsystem duplicating that effort, potentially being out of synchronization with each other. Subsystem A should be able to expect that subsystem B is working on the same relay-parents as it is. One of the Overseer's tasks is to provide this heartbeat, or synchronized rhythm, to the system.
 
 The work that subsystems spawn to be done on a specific relay-parent is known as a job. Subsystems should set up and tear down jobs according to the signals received from the overseer. Subsystems may share or cache state between jobs.
+
+Subsystems must be robust to spurious exits. The outputs of the set of subsystems as a whole comprises of signed messages and data committed to disk. Care must be taken to avoid issuing messages that are not substantiated. Since subsystems need to be safe under spurious exits, it is the expected behavior that an `OverseerSignal::Conclude` can just lead to breaking the loop and exiting directly as opposed to waiting for everything to shut down gracefully.
```

Comment:
I bet we have many places in code that are expected to do transactional writes to storage, but in practice do several sequential writes (`<key1>::insert`, `<key2>::insert`), e.g. 
https://github.com/paritytech/polkadot/pull/1413/files#diff-91e16b32e3291fe22a39b8ec31d63d97R451

is that something to worry about or am I misunderstanding the issue here? At what granularity can spurious exits happen?

Reply:
> I bet we have many places in code that are expected to do transactional writes to storage, but in practice do several sequential writes (<key1>::insert, <key2>::insert)

I don't know if we do. KeyValueDB exposes a `DBTransaction` that lets us do atomic updates.

Reply:
The point is that a spurious exit could happen anyway if the computer lost power, and we have to be resilient to that. In practice, we will wait some time for jobs to conclude, but can't rely on that as an invariant. This is why we should only sign messages attesting to availability _after_ things have been committed to the database.